interceptors/src/main/java/com/connexta/spring/interceptor/VersionInterceptor.java

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.spring.interceptor;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.security.CodeSource;
+import java.security.ProtectionDomain;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.input.ProxyInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+/**
+ * Spring interceptor capable of validating the version provided by the client with the version
+ * implemented by the server and throws a {@link UnsupportedVersionException} exception if the
+ * client version is not supported. It will also automatically add the <code>Content-Version</code>
+ * header with the server version for the API called. This only happens in cases where the
+ * controller was called and generated a response and no other interceptors failed.
+ *
+ * <p>The server version for the API is retrieved from the dependency.properties file that must be
+ * included in the generated spring API using Apache's ServiceMix depends maven plugin.
+ */
+@Component
+public class VersionInterceptor implements HandlerInterceptor {
```

Comment:
Will this class be used to help update all the endpoints?

Reply:
This class and all others in this package are a copy from the discovery-api repos. This particular class will be registered with spring to handle version automatically. Let's make sure we have a ticket to do that.

Reply:
Yes there is a ticket to do that. There is also a ticket to create the exception interceptor.